src/main/resources/avro/alleleAnnotations.avdl

```diff
@@ -0,0 +1,190 @@
+as SIFT on a specific allele.
+*/
+record AnalysisResult {
+
+  /** The analysis record for this result (defined in metadata schema)  */
+  Analysis analysis;
+
+  /** The text-based result for this analysis */
+  union { null, string} analysisResult;
+
+  /** The numeric score for this analysis  */
+  union { null, int} analysisScore;
+
+}
+
+/**
+An allele location record holds the location of an allele relative to a
+non-genomic coordinate system such as a CDS or protein and holds the
+reference and alternate sequence where appropriate
+*/
+record AlleleLocation {
+
+  /** Relative start position of the allele in this coordinate system */
+  int overlapStart ;
+
+  /** Relative end position of the allele in this coordinate system  */
+  union { null, int } overlapEnd = null;
+
+  /** Reference sequence in feature (this should be the codon at CDS level) */
+  union { null, string } referenceSequence = null;
+
+  /** Alternate sequence in feature (this should be the codon at CDS level) */
+  union { null, string } alternateSequence = null;
+}
+
+/**
+Impact is a simple prioritization for the effect of an allele which is used
+in the annotation record.
+IMPORTANT:
+ Prioritization methods are a crude estimates and are not assumed to be
+ reliable: a 'HIGH' Impact may actually not cause any disruption
+ in protein function or expression.
+*/
+enum Impact {
+ HIGH,      // Variant highly disrupts protein function
+ MODERATE,  // Moderately disrupts protein function
+ LOW,       // Low disruption of protein impact
+ MODIFIER   // No known effect
+}
+
+/**
+A VariantAnnotationSet record groups VariantAnnotation records. It is derived
+from a VariantSet and holds information describing the software and reference
+data used in the annotation.
+*/
+record VariantAnnotationSet {
+
+  /** The ID of the variant annotation set record */
+  string id;
+
+  /** The ID of the variant set to which this annotation set belongs */
+  string variantSetId;
+
+  /**
+  Analysis details. It is essential to supply versions for all software and
+  reference data used.
+  */
+  Analysis analysis;
+}
+
+/**
+A transcript effect record is a set of information describing the
+effect of an allele on a transcript
+*/
+record TranscriptEffect {
+
+  /** The ID of the transcript effect record
+  */
+  string id;
+
+  //TODO: derive unique id from digest of data [location, allele, transcript?]
+
+  /** The id of the transcript feature the annotation is relative to
+  */
+  string featureId;
+
+  /** Alternate allele - a variant may have more than one alternate allele,
+  each of which will have distinct annotation.
+  */
+  union { null, string} alternateBases = null;
+
+  /** Effect of variant on this feature */
+  array<OntologyTerm> effects;
+
+  /** Highest Impact from the predicted effects */
+  Impact impact;
+
+  /** HGVS formatted annotation at genomic level  */
+  union { null, string } HGVSg = null;
```

Comment:
Someone reading client code shouldn't have to look up the API documentation to find out what something is though --- they should be able to just look at it and know what it is. If I saw:
```python
transcriptEffect.hgvsAnnotation.genomic
```
in client code it would be much more obvious. 

It would be good to get @reece's opinion here, since he knows a lot more about HGVS than I do!

Either way, I wouldn't block on the point. It can be followed up as an issue after merge IMO.